pkg/cluster/generateconfig.go

```diff
@@ -113,7 +113,7 @@ func (m *manager) generateInstallConfig(ctx context.Context) (*installconfig.Ins
 						CIDR: *ipnet.MustParseCIDR("127.0.0.0/8"), // dummy
 					},
 				},
-				NetworkType: "OpenShiftSDN",
+				NetworkType: m.doc.OpenShiftCluster.Properties.NetworkProfile.NetworkType,
```

Comment:
Try to create a cluster not via `2021-01-31-preview` api version: I think you wil get an empty string here and cluster creation will probably fail.

Reply:
This should not be the case, as the parameter has been defaulted to "OVNKubernetes" here: https://github.com/Azure/ARO-RP/pull/1571/files#diff-c7dba7bf3908043db4c458af9edd806998129f1ef7f58ee8fa54a312ce93f23cR356 this is not API version dependent.

Reply:
@drewandersonnz the problem is that `pkg/util/cluster/cluster.go`:

1. Is not being used in production clsuter creation. We use this util in `./hack/cluster create` dev tool and in e2e cluster creation.
2. It actually is API version dependant. We populate internal representation (`api.OpenShiftCluster`) in this util (for simplicity, I guess), but then convert it to an external (API version specific) representation and fire creation to a specific API version:
    * If we are not in dev env (not sure where it is used actually), we create using v20200430. In this case the field will be droped.
      https://github.com/Azure/ARO-RP/blob/d70dc5b473e3b63f525d5c8396f685bc08cbf509/pkg/util/cluster/cluster.go#L411-L417
    * If we are in dev env, we create using v20210131preview. This applies to CI and this is why CI reports green on this RP (which is bad and which is why we need better e2e strategy)
      https://github.com/Azure/ARO-RP/blob/d70dc5b473e3b63f525d5c8396f685bc08cbf509/pkg/util/cluster/cluster.go#L396-L402

I recommend you to create a cluster using older version by either:

* Using az CLI
* Or firing an API request with `curl`. Example is [here](https://github.com/Azure/ARO-RP/pull/1569), but you don't need keyvault and disc encrpytion sets. And you need to use older API version.

Reply:
Actually... It might also be the case for the new api version: `sdnProvider` is optional (`omitempty`). So if you use the new api version, but create a cluster without providing SDN provider.

Basically these are the cases you need to take care of:

* Cluster creation with older API versions (where we can't provide `sdnProvider`)
* Cluster creation with the new API version where you omit `sdnProvider (same as removing `SDNProvider: "OVNKubernetes"` from `pkg/util/cluster/cluster.go`).
* Cluster creation with the new API version where you provide `sdnProvider`

So far the PR seem to handle only the last one.

Reply:
Sorry @m1kola you are right. This was my misunderstanding in the way the object was being handled, the generateconfig.go now handles this case.

Reply:
Created default here: https://github.com/Azure/ARO-RP/pull/1571#pullrequestreview-707279749